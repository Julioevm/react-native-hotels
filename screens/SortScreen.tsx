import {Button, StyleSheet, Text, View} from 'react-native';
import React, {useContext} from 'react';
import {Picker} from '@react-native-picker/picker';
import {Slider} from '@miblanchard/react-native-slider';
import {Theme} from '../Theme';
import {NativeStackScreenProps} from '@react-navigation/native-stack';
import {RootStackParamList} from '../types';
import FilterContext from '../context/FilterContext';

type Props = NativeStackScreenProps<RootStackParamList, 'Sort'>;

export default function SortScreen({navigation}: Props) {
  const {filters, setFilters} = useContext(FilterContext);
  const {sort, order, price} = filters;
  function onApplyPress() {
    navigation.goBack();
  }

  function onClearPress() {
    setFilters({
      name: '',
      sort: undefined,
      order: 'descending',
      price: [0, 500],
    });
    navigation.goBack();
  }

  console.log(filters);

  return (
    <View style={styles.container}>
      <View>
        <Text>Sort by:</Text>
        <Picker
          style={styles.picker}
          testID="sort-picker"
          selectedValue={sort}
          onValueChange={itemValue => {
            setFilters(prevFilters => {
              return {
                ...prevFilters,
                sort: itemValue,
              };
            });
          }}>
          <Picker.Item label="None" value={undefined} />
          <Picker.Item label="Rating" value="rating" />
          <Picker.Item label="Stars" value="stars" />
          <Picker.Item label="Price" value="price" />
        </Picker>
        <Text>Order:</Text>
        <Picker
          style={styles.picker}
          selectedValue={order}
          onValueChange={itemValue => {
            setFilters(prevFilters => {
              return {
                ...prevFilters,
                order: itemValue,
              };
            });
          }}>
          <Picker.Item label="Ascending" value="ascending" />
          <Picker.Item label="Descending" value="descending" />
        </Picker>
      </View>
      <View>
        <Text>Price Range:</Text>
        <Text>
          {price[0]} - {price[1]}
        </Text>
        <Slider
          containerStyle={styles.slider}
          value={price}
          minimumValue={0}
          maximumValue={500}
          step={1}
          onValueChange={value => {
            setFilters(prevFilters => {
              return {
                ...prevFilters,
                price: value as number[],
              };
            });
          }}
          thumbTintColor={Theme.colors.primary}
        />
      </View>
      <Button
        title="Apply"
        onPress={onApplyPress}
        color={Theme.colors.primary}
        testID="apply"
      />
      <Text style={styles.textButton} onPress={onClearPress}>
        Clear
      </Text>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    flex: 1,
    justifyContent: 'center',
  },
  picker: {
    marginVertical: 15,
    padding: 10,
    width: 250,
  },
  slider: {
    marginVertical: 15,
    padding: 10,
    width: 250,
  },
  textButton: {
    color: Theme.colors.primary,
    fontWeight: 'bold',
    margin: Theme.sizes.large,
  },
});
