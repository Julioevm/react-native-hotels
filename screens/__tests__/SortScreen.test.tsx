import {render} from '@testing-library/react-native';
import SortScreen from '../SortScreen';
import React from 'react';

describe('SortScreen', () => {
  it('should render', () => {
    const {toJSON} = render(
      //@ts-ignore ignore the navigation props
      <SortScreen navigation={undefined} route={undefined} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
