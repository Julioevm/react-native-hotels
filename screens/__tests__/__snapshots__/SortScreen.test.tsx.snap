// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortScreen should render 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View>
    <Text>
      Sort by:
    </Text>
    <View
      style={
        {
          "marginVertical": 15,
          "padding": 10,
          "width": 250,
        }
      }
    >
      <RNCPicker
        items={
          [
            {
              "label": "None",
              "testID": undefined,
              "textColor": undefined,
              "value": undefined,
            },
            {
              "label": "Rating",
              "testID": undefined,
              "textColor": undefined,
              "value": "rating",
            },
            {
              "label": "Stars",
              "testID": undefined,
              "textColor": undefined,
              "value": "stars",
            },
            {
              "label": "Price",
              "testID": undefined,
              "textColor": undefined,
              "value": "price",
            },
          ]
        }
        numberOfLines={1}
        onChange={[Function]}
        selectedIndex={0}
        style={
          [
            {
              "height": 216,
            },
            undefined,
          ]
        }
        testID="sort-picker"
      />
    </View>
    <Text>
      Order:
    </Text>
    <View
      style={
        {
          "marginVertical": 15,
          "padding": 10,
          "width": 250,
        }
      }
    >
      <RNCPicker
        items={
          [
            {
              "label": "Ascending",
              "testID": undefined,
              "textColor": undefined,
              "value": "ascending",
            },
            {
              "label": "Descending",
              "testID": undefined,
              "textColor": undefined,
              "value": "descending",
            },
          ]
        }
        numberOfLines={1}
        onChange={[Function]}
        selectedIndex={1}
        style={
          [
            {
              "height": 216,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
  <View>
    <Text>
      Price Range:
    </Text>
    <Text>
      0
       - 
      500
    </Text>
    <View
      animationType="timing"
      onLayout={[Function]}
      onValueChange={[Function]}
      step={1}
      style={
        [
          {
            "height": 40,
            "justifyContent": "center",
          },
          {},
          {
            "marginVertical": 15,
            "padding": 10,
            "width": 250,
          },
        ]
      }
      trackClickable={true}
      trackMarks={[]}
      value={
        [
          0,
          500,
        ]
      }
    >
      <View
        onLayout={[Function]}
        renderToHardwareTextureAndroid={true}
        style={
          [
            {
              "borderRadius": 2,
              "height": 4,
            },
            {
              "backgroundColor": "#b3b3b3",
            },
            undefined,
          ]
        }
      />
      <View
        collapsable={false}
        renderToHardwareTextureAndroid={true}
        style={
          {
            "backgroundColor": "#3f3f3f",
            "borderRadius": 2,
            "height": 4,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "width": 0,
          }
        }
      />
      <View
        collapsable={false}
        onLayout={[Function]}
        style={
          {
            "backgroundColor": "tomato",
            "borderRadius": 10,
            "height": 20,
            "opacity": 0,
            "position": "absolute",
            "transform": [
              {
                "translateX": 0,
              },
              {
                "translateY": 0,
              },
            ],
            "width": 20,
          }
        }
      />
      <View
        collapsable={false}
        onLayout={[Function]}
        style={
          {
            "backgroundColor": "tomato",
            "borderRadius": 10,
            "height": 20,
            "opacity": 0,
            "position": "absolute",
            "transform": [
              {
                "translateX": 0,
              },
              {
                "translateY": 0,
              },
            ],
            "width": 20,
          }
        }
      />
      <View
        onMoveShouldSetResponder={[Function]}
        onMoveShouldSetResponderCapture={[Function]}
        onResponderEnd={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderReject={[Function]}
        onResponderRelease={[Function]}
        onResponderStart={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        style={
          [
            {
              "backgroundColor": "transparent",
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            {
              "marginBottom": -20,
              "marginLeft": -20,
              "marginRight": -20,
              "marginTop": -20,
            },
          ]
        }
      />
    </View>
  </View>
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="apply"
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
            {
              "color": "tomato",
            },
          ]
        }
      >
        Apply
      </Text>
    </View>
  </View>
  <Text
    onPress={[Function]}
    style={
      {
        "color": "tomato",
        "fontWeight": "bold",
        "margin": 16,
      }
    }
  >
    Clear
  </Text>
</View>
`;
